specs/AppLifecycle/Instancing/AppLifecycle SingleMulti-Instancing.md

```diff
@@ -0,0 +1,422 @@
+expect. An app can have multiple instances available as redirection targets.
+
+-   **Platform**: Only instances that have registered a key can be a target for redirection.
+-   **Reunion**: Any instance can be a redirection target, whether or not it has a registered key.
+
+### Post-Redirection Behavior
+
+Redirection is a terminal operation. After calling the method to redirect, there is nothing the app
+can do - regardless of whether the redirect succeeded or failed. In calling redirect, the app is
+saying that it cannot/will not handle this activation request - so even if the redirection fails,
+there's nothing this instance can or will do anyway. The platform will explicitly terminate an
+instance that has chosen to redirect. If redirection fails, the activation request fails. The UX is
+the same as an activation failure today.
+
+### Unregistering
+
+At any time, an instance of the app can unregister its key. At any time, an app instance can
+re-register itself (again with any arbitrary key, not necessarily the same key each time), even
+after it has unregistered itself.
+
+-   **Platform**: When an instance unregisters itself, it is no longer available for activation
+    redirection. An app that has unregistered itself is not included in the list of instance that is
+    made available to the app.
+-   **Reunion**: Unregistering a key does not prevent an instance from being a target for
+    redirection. Nor does it exclude the instance from the list.
+
+### Activation Events
+
+In order to handle reactivation, the app can register for an Activated event.
+
+-   **Platform**: The event passes a `Windows.ApplicationModel.Activation.IActivatedEventArgs` to
+    the app.
+-   **Reunion**: The event passes a `Microsoft.Windows.AppLifecycle.ActivationArguments` instance to
+    the app, which contains one of the `-ActivatedEventArgs` instances.
+
+## Examples
+
+### Redirection logic based on activation kind
+
+In this example, the app calls GetActivatedEventArgs, and if it has been activated as a result of a
+file-type association, it checks first to see if there is any other instance already running that
+has this file open. The app can tell whether the instance has the file open because it includes the
+filename as part of the key that it registers with the system.
+
+If it finds another instance with the file open, the app redirects this activation to that instance,
+and then does nothing. After redirecting the activation to the other instance, the system will then
+terminate this instance.
+
+If it doesn't find another instance with the file open, it registers itself as the instance that has
+this file open, and continues with normal app initialization (creating its main window, starting its
+message pump, and so on).
+
+```c++
+int APIENTRY wWinMain(
+    _In_ HINSTANCE hInstance, _In_opt_ HINSTANCE hPrevInstance,
+    _In_ LPWSTR lpCmdLine, _In_ int nCmdShow)
+{
```

Comment:
Do we need to CoInitialize? Or can these methods operate without COM?

Reply:
Fixed.